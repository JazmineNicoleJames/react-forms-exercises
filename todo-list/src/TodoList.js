import React, {useState} from "react";
import { v4 as uuid } from "uuid";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo"


const TodoList = () => {

    const [todo, setTodo] = useState([])
    const [editTodo, setEditTodo] = useState(null)
    
    const remove = (id) => {
        setTodo(((prevTodo) => prevTodo.filter(todo => todo.id !==id)))
    }

    const addTodo = (todo) => {
        const id = uuid()
        let newTodo = {
            ...todo,
            id
        }
        setTodo((prevTodo) => [...prevTodo, newTodo]);
        setEditTodo(null)
    };

    const clickEditTodo = (id) => {
        const todoEdit = todo.find((todo) => todo.id === id)
        setEditTodo({...todoEdit, todo: todoEdit.todo})
    };

    const updateTodo = (updatedTodo) => {
        setTodo((prevTodo) => {
            return prevTodo.map((todo) => {
                if(todo.id === updatedTodo.id) {
                    setEditTodo(null)
                    return updatedTodo
                } else {
                    return todo;
                }

            });

        })
    }

    return (
        <div>
            <NewTodoForm 
                addTodo={addTodo} 
                className="NewTodoForm" 
                remove={remove} 
                editTodo={editTodo} 
                updateTodo={updateTodo} />
            <div className="TodoList">
                {todo.map(({id, todo}) =>
                    <Todo id={id} 
                        todo={todo}
                        remove={remove}
                        clickEditTodo={clickEditTodo}
                    />)}
            </div>
        </div>
    )
}

export default TodoList;