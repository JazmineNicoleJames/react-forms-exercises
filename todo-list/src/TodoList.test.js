import TodoList from "./TodoList";
import { render, fireEvent} from "@testing-library/react";

it("renders without crashing", () => {
    render(<TodoList />)
});

it("should match snapshot", () => {
    const { asFragment } = render(<TodoList />)

    expect(asFragment()).toMatchSnapshot();
});

it("should add new box", () => {
    const { queryByText, getByLabelText, getByTestId } = render(<TodoList />);

    const todoInput = getByLabelText("Todo:");
    const submitBtn = queryByText("Add todo");

    fireEvent.change(todoInput, {target: {value: "shopping"}})
    fireEvent.click(submitBtn);

    const newBox = getByTestId('todo-component');

    expect(newBox).toBeInTheDocument();
});
