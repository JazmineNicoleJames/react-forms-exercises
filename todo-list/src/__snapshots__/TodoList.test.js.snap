// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class="NewTodoForm"
      data-testid="new-todo-form"
    >
      <label
        for="todo"
      >
        Todo:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
    <div
      class="TodoList"
    />
  </div>
</DocumentFragment>
`;
